exercises/02-publish-subscribe/after/Divergent.Finance/Host.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Threading.Tasks;
+using Divergent.Finance.PaymentClient;
+using Divergent.Sales.Messages.Events;
+using ITOps.EndpointConfig;
+using NServiceBus;
+using NServiceBus.Logging;
+
+namespace Divergent.Finance
+{
+    class Host
+    {
+        static readonly ILog Log = LogManager.GetLogger<Host>();
+        readonly string connectionString;
+        IEndpointInstance endpoint;
+
+        public static string EndpointName => "Divergent.Finance";
+
+        public Host(string connectionString) => this.connectionString = connectionString;
+
+        public async Task Start()
+        {
+            try
+            {
+                var endpointConfiguration = new EndpointConfiguration(EndpointName)
+                    .Configure(
+                        connectionString,
+                        routing => routing.RegisterPublisher(typeof(OrderSubmittedEvent), "Divergent.Sales"));
+
+                endpointConfiguration.RegisterComponents(registration =>
```

Comment:
Seperate components are created so responsibilities aren't mixed, but then this mixes DI with the host and breaks SRP again?

Reply:
Same response as [previous](https://github.com/Particular/Workshop/pull/210/files#r159937815).